osquery/config/config.cpp

```diff
@@ -194,21 +194,22 @@ void Schedule::remove(const std::string& pack) {
 
 void Schedule::remove(const std::string& pack, const std::string& source) {
   packs_.remove_if([pack, source](PackRef& p) {
-    if (p->getName() == pack && (p->getSource() == source || source == "")) {
+    if (p && p->getName() == pack &&
+        (p->getSource() == source || source == "")) {
       Config::get().removeFiles(source + FLAGS_pack_delimiter + p->getName());
-      return true;
+      p.reset();
```

Comment:
nit: is the p.reset() needed here? I suppose the unique_ptr will be removed and therefore the contained ptr freed, no?

Reply:
I see your point. I didn't assume that p could ever be empty. Is there a reason to think that might be the case? At any rate lets keep the code as is.

Reply:
Yep, you are right. I'll remove it. Thanks!